### PR TITLE
Puts a cap on butterflies spawned from the Severe Anxiety disease

### PR DESCRIPTION
**Pull request by Altoids1**
## Holy Shit why are there 500 dead butterfly corpses
image

So a few days back there was a highpop round where this Anxiety disease spread to just about everybody. 45 minutes later into the round, we had people (especially those on potato computers) lagging out hard or even having to disconnect, due to the 500-odd butterflies which had infested basically every tile of Medbay and the populous hallways.

This PR puts a cap on the total amount of butterflies that can be spawned by this disease,
 at a reasonable **250**. Hopefully this will help improve the performance of this.

#### Changelog
:cl:  Altoids
bugfix: The Anxiety disease can no longer spawn an infinite number of butterflies. Just 250.
/:cl:

### CODE REVIEW

**Review by Amelia0010 — CHANGES_REQUESTED** [review]
make butterflies dying bring the count down again.

**Comment by Altoids1** [comment]
Nich why would there need to be more than 250 butterflies on the station, dead or alive

**Comment by Amelia0010** [comment]
because otherwise it doesn't do butterflies anymore ;-;

**Comment by ThatLing** [comment]
just kill the old ones when a new one spawns

File: code/datums/diseases/anxiety.dm [hunk 1]
```diff
@@ -36,6 +36,17 @@
 			if(prob(2))
 				affected_mob.visible_message("<span class='danger'>[affected_mob] coughs up butterflies!</span>", \
 													"<span class='userdanger'>You cough up butterflies!</span>")
-				new /mob/living/simple_animal/butterfly(affected_mob.loc)
-				new /mob/living/simple_animal/butterfly(affected_mob.loc)
+				//Yogs start -- Puts a maximum on the number of butterflies that can be spawned via this disease
+				var/mob/living/simple_animal/butterfly/Alpha = new(affected_mob.loc)
+				var/mob/living/simple_animal/butterfly/Beta = new(affected_mob.loc)
+				butterfly_count += list(Alpha,Beta)
+				if(butterfly_count.len > 250)
```
Comment (Kmc2000) [inline code/datums/diseases/anxiety.dm:43]: This seems like an ultra clunky way to do it.
Could you not simply add a reference to this disease to the butterfly mobs, and then on death() have it remove itself from the list?

File: yogstation/code/datums/diseases/anxiety.dm [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/datum/disease/anxiety
+	var/static/list/butterfly_count = list() // Counts how many butterflies have been spawned by this disease
```
Comment (Kmc2000) [inline yogstation/code/datums/diseases/anxiety.dm:2]: Why static?

File: code/datums/diseases/anxiety.dm [hunk 3]
```diff
@@ -36,6 +36,17 @@
 			if(prob(2))
 				affected_mob.visible_message("<span class='danger'>[affected_mob] coughs up butterflies!</span>", \
 													"<span class='userdanger'>You cough up butterflies!</span>")
-				new /mob/living/simple_animal/butterfly(affected_mob.loc)
-				new /mob/living/simple_animal/butterfly(affected_mob.loc)
+				//Yogs start -- Puts a maximum on the number of butterflies that can be spawned via this disease
+				var/mob/living/simple_animal/butterfly/Alpha = new(affected_mob.loc)
```
Comment (Kmc2000) [inline code/datums/diseases/anxiety.dm:40]: I don't get the point of this

**Pull request closed without merging**